seahub/admin_log/models.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2012-2017 Seafile Ltd.
+import json
+import datetime
+
+from django.db import models
+from django.db.models import Q
+from django.dispatch import receiver
+
+from seahub.admin_log.signals import admin_operation
+
+ADMIN_LOG_OPERATION_TYPE = ('repo_transfer', 'repo_delete',
+        'group_create', 'group_transfer', 'group_delete',
+        'user_add', 'user_delete')
+
+## operation: detail
+
+# 'repo_transfer': {'id': repo_id, 'name': repo_name, 'from': from_user, 'to': to_user}
+# 'repo_delete': {'id': repo_id, 'name': repo_name, 'owner': repo_owner}
+
+# 'group_create': {'id': group_id, 'name': group_name, 'owner': group_owner}
+# 'group_transfer': {'id': group_id, 'name': group_name, 'from': from_user, 'to': to_user}
+# 'group_delete': {'id': group_id, 'name': group_name, 'owner': group_owner}
+
+# 'user_add': {'email': new_user}
+# 'user_delete': {'email': deleted_user}
+
+
+class AdminLogManager(models.Manager):
+
+    def add_admin_log(self, email, operation, detail):
+
+        model= super(AdminLogManager, self).create(
+            email=email, operation=operation, detail=detail)
+
+        model.save()
+
+        return model
+
+    def get_admin_logs(self, email=None, operation=None):
+
+        logs = super(AdminLogManager, self).all()
+
+        if email and operation:
```

Comment:
if email is not None: logs = logs.filter(..)
if operation is not None : logs = ...